src/debug.cc

```diff
@@ -487,21 +985,37 @@ Debug::parseOptions(char const *options)
 }
 
 void
-_db_init(const char *logfile, const char *options)
+Debug::BanCacheLogUse()
 {
-    Debug::parseOptions(options);
+    Debug::parseOptions(debugOptions);
+    Module().banCacheLogUse();
+}
 
-    debugOpenLog(logfile);
+void
+Debug::UseCacheLog()
+{
+    Debug::parseOptions(debugOptions);
+    debugOpenLog(cache_log);
+}
+
+void
+Debug::StopCacheLogUse()
+{
+    TheLog.clear(); // may already be closed
+    Module().cacheLogChannel.stopEarlyMessageCollection(); // may already be stopped
```

Comment:
It looks like we should log the early messages first and then close the log.

Reply:
Good point. I will adjust the method description to allow buffer flushing and swap the two lines above.

Reply:
Addressed in f59ce0d. Please review that commit to make sure we are on the same page.

FYI: Simply reordering the two code lines probably makes no difference because stopping early message collection does not magically redirect any saved messages to another channel. The channel will just attempt to log them to that channel stream. At this point,

* in the current code, the cache_log channel should have no saved messages because it is writing to cache_log or failed to open that cache_log and stopped saving messages at that time.
* In the future, the cache_log channel may have saved messages but no cache_log file to log those saved messages to (because we did not _try_ to open cache_log yet). New commit f59ce0d redirects cache_log to stderr in this hypothetical future case.